tests/test_units.py

```diff
@@ -20,14 +21,19 @@ def test_add_unit(self):
         # Add ordinary unit
         unitstore = UnitStore()
         assert not unitstore.is_defined('u1')
-        unitstore.add_unit('u1', 'second * 2')
+        u1 = unitstore.add_unit('u1', 'second * 2')
         assert unitstore.is_defined('u1')
-        assert unitstore.get_unit('u1') == unitstore.get_unit('second') * 2
+        assert u1 == unitstore.get_unit('second') * 2
```

Comment:
Maybe also `assert u1 is unitstore.get_unit('u1')`?

Reply:
Weirdly, it has to be `==`. Repeated calls to `get_unit()` don't return the same object.